Signal/src/environment/Migrations/OWS105AttachmentFilePaths.m

```diff
@@ -0,0 +1,58 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWS105AttachmentFilePaths.h"
+#import <SignalServiceKit/TSAttachmentStream.h>
+#import <YapDatabase/YapDatabaseTransaction.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+// Increment a similar constant for every future DBMigration
+static NSString *const OWS105AttachmentFilePathsMigrationId = @"105";
+
+@implementation OWS105AttachmentFilePaths
+
++ (NSString *)migrationId
+{
+    return OWS105AttachmentFilePathsMigrationId;
+}
+
+- (void)runUpWithTransaction:(YapDatabaseReadWriteTransaction *)transaction
+{
+    OWSAssert(transaction);
+
+    NSMutableArray<TSAttachmentStream *> *attachmentStreams = [NSMutableArray new];
+    [transaction enumerateKeysAndObjectsInCollection:TSAttachmentStream.collection
+                                          usingBlock:^(NSString *key, TSAttachment *attachment, BOOL *stop) {
+                                              if (![attachment isKindOfClass:[TSAttachmentStream class]]) {
+                                                  return;
+                                              }
+                                              TSAttachmentStream *attachmentStream = (TSAttachmentStream *)attachment;
+                                              if (attachmentStream.hasUnsavedFilePath) {
```

Comment:
Proposal:

In ssk, rename `ensureFilePathAndPersist:` to just `ensureFilePath`, and remove any of the persistence logic.

Then change this migration to:
```
[attachmentStream ensureFilePath];
[attachmentStream saveWithTransaction:transaction];
```

We continue to call `ensureFilePath` in initWithCoder, which will be fast because of this migration.

That would let us get rid of tracking this new `hasUnsavedFilePath` property which is only used here, in this migration and allows us to simplify the `ensureFilePath` method.

WDYT @charlesmchen?